### PR TITLE
Rewrite model check as validation check re #10079

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
When trying my hand at writing a django system check in #10079, I missed a pretty major [caveat](https://docs.djangoproject.com/en/5.1/topics/checks/#writing-your-own-checks), which is that if you use the database, you need to do something like:

https://github.com/archesproject/arches/blob/0f23a6cbc78a093e5a08cf2ab2abf2c542915df4/arches/app/models/models.py#L619-L620

... to avoid db hits during commands that don't generally expect a database to be configured. (My earlier workaround was was to run subsets of checks in different places, but that was kind of a code smell. This PR removes some of those workarounds.)

But to start observing `kwargs["databases"]` means that these checks only run on:
```
- manage.py migrate
- manage.py check --database default
```

So, given how infrequently this would be run, better to just implement this in our validation framework and call it out in the release notes?

#### Testing instructions
- Create and publish a graph
- Enable an additional system language, without republishing any graphs
- manage.py validate
- expect errros
- manage.py validate --fix
- expect errrors fixed
- ensure branches do not trigger the check
- ensure languages for which a publication exists but is not currently in settings.LANGUAGES doesn't alter the comparison and cause the check to unexpectedly pass

### CODE REVIEW

File: arches/management/commands/validate.py [hunk 1]
```diff
@@ -116,6 +120,36 @@ def handle(self, *args, **options):
             ),
             fix_action=DELETE_QUERYSET,
         )
+        self.check_integrity(
+            check=IntegrityCheck.PUBLICATION_MISSING_FOR_LANGUAGE,  # 1014
+            queryset=(
+                models.GraphModel.objects.filter(
+                    isresource=True,
```
Comment (jacobtylerwalls) [inline arches/management/commands/validate.py:127]: The prior version of this was also checking branches when it shouldn't.

File: arches/app/models/models.py [hunk 2]
```diff
@@ -613,39 +613,6 @@ def save(self, *args, **kwargs):
 
         super(GraphModel, self).save(*args, **kwargs)
 
-    @classmethod
```
Comment (jacobtylerwalls) [inline arches/app/models/models.py:616]: Pretty sure this deletion is the main reason the coverage check is failing.

**Comment by jacobtylerwalls** [comment]
> So, given how infrequently this would be run, better to just implement this in our validation framework and call it out in the release notes?

Some more info about the original motivation here: https://github.com/archesproject/arches/pull/11342#issuecomment-2491470794

**Review by aarongundel — APPROVED** [review]
LGTM

**Pull request merged**